tests/ui/regions/desktop/categories.py

```diff
@@ -0,0 +1,33 @@
+from pypom import Region
+from selenium.webdriver.common.by import By
+
+
+class Categories(Region):
+
+    _root_locator = (By.CLASS_NAME, 'Categories')
+    _categories_locator = (By.CLASS_NAME, 'Categories-item')
+    _mobile_categories_locator = (By.CLASS_NAME, 'LandingPage-button')
+
+    @property
+    def category_list(self):
+        # If it's a mobile page, return the mobile list which is different
+        window = self.selenium.get_window_size()
+        if int(window['width']) < 500:
+            self.selenium.find_element(
+                *self._mobile_categories_locator).click()
+            from pages.mobile.categories import Categories
+            return Categories(self.selenium, self.page).category_list
+        items = self.find_elements(*self._categories_locator)
+        return [self.CategoryList(self, el) for el in items]
```

Comment:
As noticed earlier, this should be explicitly requested by the respective test to be either this or that, e.g by defining `category_list` and `category_list_mobile` so that the test switches logic based on window-sizes. This way we get much more reasonable failures in case the base configuration (e.g parametrization) changes.

Reply:
Ah, now I see what you mean. I may have been trying to be a bit fancy.

Reply:
Would make more sense to return an error specific to the size if something does fail? Since the tests are parametrized, the current implementation of this works well and would be difficult to change.